dataactcore/interfaces/function_bag.py

```diff
@@ -118,3 +119,125 @@ def checkNumberOfErrorsByJobId(jobId, errorType='fatal'):
     # returns an empty value that means the job didn't have any errors that matched
     # the specified severity type, so return 0
     return errors or 0
+
+
+def addJobsForFileType(fileType, filePath, filename, submissionId, existingSubmission, jobsRequired, uploadDict):
+    """ Add upload and validation jobs for a single filetype
+
+    Args:
+        fileType: What type of file to add jobs for
+        filePath: Path to upload the file to
+        filename: Original filename
+        submissionId -- Submission ID to attach to jobs
+        existingSubmission -- True if we should update existing jobs rather than creating new ones
+        jobsRequired: List of job ids that will be prerequisites for cross-file job
+        uploadDict: Dictionary of upload ids by filename to return to client, used for calling finalize_submission route
+
+    Returns:
+        jobsRequired: List of job ids that will be prerequisites for cross-file job
+        uploadDict: Dictionary of upload ids by filename to return to client, used for calling finalize_submission route
+    """
+    sess = GlobalDB.db().session
+
+    fileTypeId = FILE_TYPE_DICT[fileType]
+
+    # Create a file upload job or, for an existing submission, modify the
+    # existing upload job.
+
+    if existingSubmission:
+        # mark existing upload job as running
+        uploadJob = sess.query(Job).filter(Job.submission_id == submissionId).\
+            filter(Job.file_type_id == fileTypeId, Job.job_type_id == JOB_TYPE_DICT['file_upload']).one()
```

Comment:
Well, that is very lovely. Seems to work as expected, so I added it here and in one other place to get in the habit.